packages/simplebar/src/simplebar.js

```diff
@@ -752,6 +753,12 @@ export default class SimpleBar {
 
     document.addEventListener('mousemove', this.drag, true);
     document.addEventListener('mouseup', this.onEndDrag, true);
+    if (this.removePreventClickId !== null) {
```

Comment:
Is this necessary at this step? That should happen only on drag, as it's user's event, in theory that's impossible that the `timeout` didn't execute in between two UI events, right?

However what is possible is that the element gets removed from the DOM, so I think we need the clearTimeout in the [removeEventListeners](https://github.com/Grsmto/simplebar/blob/86ebe41372f71481240b4a50b560c608df4c5c61/packages/simplebar/src/simplebar.js#L854) function.

Reply:
> Is this necessary at this step? That should happen only on drag, as it's user's event, in theory that's impossible that the `timeout` didn't execute in between two UI events, right?

I don’t know for sure. There are a number of [reasons](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout#Reasons_for_delays_longer_than_specified) that timer events could be delayed, and also no guarantees that input events won’t be batched and delivered “simultaneously”, so I wanted to be safe.

> However what is possible is that the element gets removed from the DOM, so I think we need the clearTimeout in the [removeEventListeners](https://github.com/Grsmto/simplebar/blob/86ebe41372f71481240b4a50b560c608df4c5c61/packages/simplebar/src/simplebar.js#L854) function.

This is a good scenario to think about, but I think my conclusion is different. If the element winds up being removed from the DOM during a drag operation—either before, or somehow immediately in response to, its end—I claim we still want to consider the mouse pointer to be captured and suppress click-like events on other elements until just after the user releases the button. They clicked on the scrollbar, and they’re not expecting that click to be reinterpreted somewhere else. Clearing the timeout would leave the `ignoreClick` handler installed forever, which would definitely be wrong, and clearing the timeout while also uninstalling the `ignoreClick` handler synchronously seems pointless and potentially wrong. So I think the current PR is correct here.

Reply:
Right. I think there is still a scenario where the `addEventListener` could stack up tho no?
So if there is a batch input of drag events for example, here you would `clearTimeout` and `this.removePreventClickId = null` but the event listeners would still be there as the callback hasn't been called.
So I suppose the `setTimeout` callback should be as a separate function and be called there, instead of doing just `this.removePreventClickId = null;`?

Reply:
These don’t stack because `addEventListener` [discards](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#Multiple_identical_event_listeners) duplicate listeners. However, it’s easy enough to avoid relying on this by conditionalizing the `addEventListener` calls on `removePreventClickId === null`—otherwise we know the listeners were already installed—so I’ve done that.